### ISSUE TITLE
Can not change Latex Color

### ISSUE THREAD

**Issue by rickalex21** [xournalpp/xournalpp#6164]
### Operating System

Linux

### (Linux only) Distribution

nixos

### (Linux only) Desktop Environment

sway

### (Linux Only) Display Server

Wayland

### Installation Method

package nixos

### Xournal++ Version

1.2.5

### libgtk Version

3.24.43

### Bug Description

When doing latex it stays on random colors and I can not change it. This
is a temp fix here: https://unix.stackexchange.com/questions/718317/xournal-does-not-change-the-color-of-latex-equation

Here I am on white and it's blue trying to tap on any other color has no effect.

![image](https://github.com/user-attachments/assets/31c80d9d-b586-44e5-8861-142d077e055c)

### Expected Behaviour

For latex to change colors when you select a color from the toolbar.

### Steps to Reproduce

1. Enter a latex formula
2. Try changing the color

The color does not change when you select a color from the tool bar.

### Additional Context

You can find a temp solution here, which describes the same problem I am facing: https://unix.stackexchange.com/questions/718317/xournal-does-not-change-the-color-of-latex-equation

**Comment by MiltonBalaOfficial** [comment]
>When doing latex it stays on random colors and I can not change it.

LaTex starts with the text color, if you want it starts with the current color you may try this [LaTexWithCurrentColor](https://github.com/MiltonBalaOfficial/LaTexWithCurrentColor)

**Comment by rickalex21** [comment]
@MiltonBalaOfficial  On other documents that may be the case that it starts with the text color but on this particual document it stays on blue no matter what I do.



https://github.com/user-attachments/assets/ed930a78-e144-4c33-8a74-6afc08e85680

**Comment by MiltonBalaOfficial** [comment]
You are changing the color of pen, first select the `text` tool, then change the color, finally start LaTex

**Comment by rickalex21** [comment]
@MiltonBalaOfficial  That's not very intuitive. Hence, others have asked the same question. However, I do appreciate the tip. Thank you. :)

**Comment by immi-h** [comment]
I'm thinking about fixing this, but in which way? One issue that makes it difficult to make this intuitive is that currently, the "insert latex" button works differently than any other tool.
Idea/suggestion: Make it work like the "insert image" tool:
- It can be chosen like the other tools. (It does not open the latex edit box right away.)
- To actually create a latex element, one has to click into the document.
- When the latex tool is active, the color that will be used (which is the same as for the text tool) is selected and can also be changed.
Edit: Having just looked at the code, it might be easier to implement that the latex tool has its own color (rather than sharing it with the text tool).

**Comment by bhennion** [comment]
I agree that the fact the LaTex tool behaves differently from other tools is the source of the issue.
It may be better to have it behave like the Text tool rather, in that when clicking on a pre-existing Tex element, it edits it rather than creating a new one.

It also sound better to have the Tex tool have its own color rather than copying the Text colors.
